python/feathub/processors/flink/table_builder/flink_table_builder.py

```diff
@@ -391,19 +393,36 @@ def _get_table_from_derived_feature_view(
             right_table_name,
             keys,
         ), right_table_join_field_descriptors in right_tables.items():
-            right_table = table_by_names[right_table_name].select(
-                *[
-                    native_flink_expr.col(right_table_field)
-                    for right_table_field in right_table_join_field_descriptors.keys()
-                ]
-            )
-            tmp_table = temporal_join(
-                self.t_env,
-                tmp_table,
-                right_table,
-                keys,
-                right_table_join_field_descriptors,
-            )
+            right_table = table_by_names[right_table_name]
+
+            if EVENT_TIME_ATTRIBUTE_NAME in right_table_join_field_descriptors.keys():
+                right_table = right_table.select(
+                    *[
+                        native_flink_expr.col(x)
+                        for x in right_table_join_field_descriptors.keys()
+                    ]
+                )
+
+                tmp_table = temporal_join(
+                    self.t_env,
+                    tmp_table,
+                    right_table,
+                    keys,
+                    right_table_join_field_descriptors,
+                )
+            else:
+                # TODO: generalize key_expr function and decouple flink table builder
+                #  from redis.
+                tmp_table, key_fields = pre_lookup_join_on_redis_table(
```

Comment:
Would it be more readable to invoke a method named `append_redis_key_columns` when the right table contains RedisSource?